### PR TITLE
feat: P4ADEV-1542 reportingflow ingestion activity implementation field validation

### PR DESCRIPTION
**Pull request by macacia**
#### How Has This Been Tested?

- Pre-Deploy Test
    - [x] Unit
    - [ ] Integration (Narrow)
- Post-Deploy Test
    - [ ] Isolated Microservice
    - [ ] Broader Integration
    - [ ] Acceptance
    - [ ] Performance & Load

#### Types of changes

- [ ] PATCH - Bug fix (backwards compatible bug fixes)
- [x] MINOR - New feature (add functionality in a backwards compatible manner)
- [ ] MAJOR - Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImpl.java [hunk 1]
```diff
@@ -52,7 +61,12 @@ public PaymentsReportingIngestionFlowFileActivityResult processFile(Long ingesti
 			CtFlussoRiversamento ctFlussoRiversamento = flussoRiversamentoUnmarshallerService.unmarshal(ingestionFlowFile);
 			log.debug("file CtFlussoRiversamento with Id {} parsed successfully ", ctFlussoRiversamento.getIdentificativoFlusso());
 
-			return new PaymentsReportingIngestionFlowFileActivityResult(List.of(ctFlussoRiversamento.getIdentificativoFlusso()), true);
+			flowValidatorService.validateOrganization(ctFlussoRiversamento, ingestionFlowFileDTO);
+
+			List<PaymentsReportingDTO> dtoListToSave = paymentsReportingMapperService.mapToDtoList(ctFlussoRiversamento, ingestionFlowFileDTO);
+
+			String iuf = dtoListToSave.get(0).getFlowIdentifierCode();
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImpl.java:68]: let's read it from ctFlussoRiversamento, where it's a single value, reading it as the first from a List could be misunderstood

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	private String receiverOrganizationId;
+	private String receiverOrganizationName;
+	private long totalPayments;
+	private BigDecimal sumPayments;
+	private String creditorReferenceId;
+	private String regulationId;
+	private BigDecimal amountPaid;
+	private String paymentOutcomeCode;
+	private Date payDate;
+	private Date acquiringDate;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:43]: let's use new date/time class
```suggestion
	private LocalDate payDate;
	private LocalDate acquiringDate;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	private String regulationUniqueIdentifier;
+	private Date regulationDate;
+	private String senderPspType;
+	private String senderPspCode;
+	private String senderPspName;
+	private String receiverOrganizationType;
+	private String receiverOrganizationId;
+	private String receiverOrganizationName;
+	private long totalPayments;
+	private BigDecimal sumPayments;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:37]: let's store it as euro cents
```suggestion
	private Long sumPaymentsCents;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	private String creditorReferenceId;
+	private String regulationId;
+	private BigDecimal amountPaid;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:40]: Let's store it as euro cents
```suggestion
	private Long amountPaidCents;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	private int version;
+	private Timestamp creationDate;
+	private Timestamp lastChangeDate;
+	private OrganizationDTO orgId;
+	private IngestionFlowFileDTO ingestionFlowFile;
+	private String idPsp;
+	private String flowIdentifierCode;
+	private Date flowDateTime;
+	private String regulationUniqueIdentifier;
+	private Date regulationDate;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:29]: let's use new date/time class
```suggestion
	private LocalDate regulationDate;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+public class PaymentsReportingDTO {
+
+	private int version;
+	private Timestamp creationDate;
+	private Timestamp lastChangeDate;
+	private OrganizationDTO orgId;
+	private IngestionFlowFileDTO ingestionFlowFile;
+	private String idPsp;
+	private String flowIdentifierCode;
+	private Date flowDateTime;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:27]: let's use new date/time class
```suggestion
	private LocalDateTime flowDateTime;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class PaymentsReportingDTO {
+
+	private int version;
+	private Timestamp creationDate;
+	private Timestamp lastChangeDate;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:22]: let's use new date/time class
```suggestion
	private Instant creationDate;
	private Instant lastChangeDate;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+	private OrganizationDTO orgId;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:23]: let's just store the id, not the relation if not needed to retrieve together with the entity
let's also keep aligned id field names
```suggestion
	private Long organizationId;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	private IngestionFlowFileDTO ingestionFlowFile;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:24]: let's just store the id, not the relation if not needed to retrieve together with the entity
```suggestion
	private Long ingestionFlowFileId;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.Date;
+
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class PaymentsReportingDTO {
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:18]: doesn't this entity have an id field?

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+	private String idPsp;
+	private String flowIdentifierCode;
+	private Date flowDateTime;
+	private String regulationUniqueIdentifier;
+	private Date regulationDate;
+	private String senderPspType;
+	private String senderPspCode;
+	private String senderPspName;
+	private String receiverOrganizationType;
+	private String receiverOrganizationId;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:34]: Is this our inner id (a Long) or an external identifier?

File: src/main/java/it/gov/pagopa/payhub/activities/service/FlowValidatorService.java [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service class responsible for validating flow data, specifically ensuring that the organization
+ * details in the flow match the expected organization.
+ */
+@Lazy
+@Service
+public class FlowValidatorService {
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/FlowValidatorService.java:15]: This is specialized on PaymentsReporting
```suggestion
public class PaymentsReportingIngestionFlowFileValidatorService {
```

File: src/main/java/it/gov/pagopa/payhub/activities/service/FlowValidatorService.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
+package it.gov.pagopa.payhub.activities.service;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/FlowValidatorService.java:1]:
```suggestion
package it.gov.pagopa.payhub.activities.service.paymentsreporting;
```

File: src/main/java/it/gov/pagopa/payhub/activities/exception/InvalidFlowDataException.java [hunk 14]
```diff
@@ -0,0 +1,8 @@
+package it.gov.pagopa.payhub.activities.exception;
+
+public class InvalidFlowDataException extends ActivitiesException{
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/exception/InvalidFlowDataException.java:3]:
```suggestion
public class InvalidIngestionFlowFileDataException extends ActivitiesException{
```

File: src/main/java/it/gov/pagopa/payhub/activities/exception/InvalidFlowDataException.java [hunk 15]
```diff
@@ -0,0 +1,8 @@
+package it.gov.pagopa.payhub.activities.exception;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/exception/InvalidFlowDataException.java:1]: let's start to partitionate exception package
```suggestion
package it.gov.pagopa.payhub.activities.exception.ingestionflow;
```

File: src/test/java/it/gov/pagopa/payhub/activities/service/FlowValidatorServiceTest.java [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+		CtIdentificativoUnivocoPersonaG ctIdentificativoUnivocoPersonaG = new CtIdentificativoUnivocoPersonaG();
+		ctIdentificativoUnivocoPersonaG.setCodiceIdentificativoUnivoco("80010020011");
+		CtIstitutoRicevente istitutoRicevente = new CtIstitutoRicevente();
+		istitutoRicevente.setIdentificativoUnivocoRicevente(ctIdentificativoUnivocoPersonaG);
+		ctFlussoRiversamento = new CtFlussoRiversamento();
+		ctFlussoRiversamento.setIstitutoRicevente(istitutoRicevente);
+	}
+
+	@Test
+	void givenDataWhenValidateThenSuccess() {
```
Comment (antonioT90) [inline src/test/java/it/gov/pagopa/payhub/activities/service/FlowValidatorServiceTest.java:32]:
```suggestion
	void givenValidDataWhenValidateThenSuccess() {
```

File: src/test/java/it/gov/pagopa/payhub/activities/service/FlowValidatorServiceTest.java [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+			.org(OrganizationDTO.builder()
+				.orgFiscalCode("80010020011")
+				.build())
+			.build();
+		// when then
+		assertDoesNotThrow(() -> service.validateOrganization(ctFlussoRiversamento, ingestionFlowFileDTO));
+	}
+
+	@Test
+	void givenDataWhenValidateThenThrowInvalidFlowDataException() {
```
Comment (antonioT90) [inline src/test/java/it/gov/pagopa/payhub/activities/service/FlowValidatorServiceTest.java:44]:
```suggestion
	void givenInvalidOrganizationWhenValidateThenThrowInvalidIngestionFlowFileDataException() {
```

File: src/test/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperServiceTest.java [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+		assertEquals("Org Ricevente", result.get(0).getReceiverOrganizationName());
+		assertEquals(100L, result.get(0).getSumPayments().longValue());
+		assertEquals(200.00d, result.get(0).getAmountPaid().doubleValue());
+		assertEquals("vers123", result.get(0).getCreditorReferenceId());
+		assertEquals("ris123", result.get(0).getRegulationId());
+		assertEquals(1, result.get(0).getTransferIndex());
+		assertEquals(200.00D, result.get(0).getAmountPaid().doubleValue());
+		assertEquals("OK", result.get(0).getPaymentOutcomeCode());
+		assertNotNull(result.get(0).getPayDate());
+	}
```
Comment (antonioT90) [inline src/test/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperServiceTest.java:85]: let's copy TestUtils.checkNotNullFields from other applications (eg. p4pa-auth) in order to check that all expected not null fields are filled

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java (hunk 11 above)
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:34]: Is an identification code like the fiscal code

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java (hunk 10 above)
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java:18]: I started from the legacy structure but @marcomatteuccieng added it in the new one, I will add it

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+import java.time.Instant;
+import java.time.LocalDate;
+
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class PaymentsReportingDTO {
+
+	private Long PaymentsReportingId;
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java]:
```suggestion
	private Long paymentsReportingId;
```

File: src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java (hunk 6 above)
Comment (antonioT90) [reply src/main/java/it/gov/pagopa/payhub/activities/dto/paymentsreporting/PaymentsReportingDTO.java]: This is a LocalDate or a LocalDateTime? I suggested LocalDateTime based on field name, but you should check each field if should contains also the time or just the date

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+	 *                             including details for multiple individual payments.
+	 * @param ingestionFlowFileDTO the ingestion metadata containing information about the processing flow.
+	 * @return a list of `PaymentsReportingDTO` objects, one for each individual payment in the flow.
+	 */
+	public List<PaymentsReportingDTO> mapToDtoList(CtFlussoRiversamento ctFlussoRiversamento, IngestionFlowFileDTO ingestionFlowFileDTO) {
+
+		PaymentsReportingDTO.PaymentsReportingDTOBuilder builder = PaymentsReportingDTO.builder()
+			.creationDate(Instant.ofEpochMilli(System.currentTimeMillis()))
+			.lastUpdateDate(Instant.ofEpochMilli(System.currentTimeMillis()))
+			.acquiringDate(new Date(ingestionFlowFileDTO.getCreationDate().getTime()).toLocalDate())
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:35]: check if the field contains just date or date/time and fix Date to LocalDate/LocalDateTime conversion
https://www.baeldung.com/java-date-to-localdate-and-localdatetime

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+	 * @param ctFlussoRiversamento the flow data object containing detailed information about the transaction flow,
+	 *                             including details for multiple individual payments.
+	 * @param ingestionFlowFileDTO the ingestion metadata containing information about the processing flow.
+	 * @return a list of `PaymentsReportingDTO` objects, one for each individual payment in the flow.
+	 */
+	public List<PaymentsReportingDTO> mapToDtoList(CtFlussoRiversamento ctFlussoRiversamento, IngestionFlowFileDTO ingestionFlowFileDTO) {
+
+		PaymentsReportingDTO.PaymentsReportingDTOBuilder builder = PaymentsReportingDTO.builder()
+			.creationDate(Instant.ofEpochMilli(System.currentTimeMillis()))
+			.lastUpdateDate(Instant.ofEpochMilli(System.currentTimeMillis()))
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:34]:
```suggestion
			.creationDate(Instant.now()
			.lastUpdateDate(Instant.now())
```

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+			.acquiringDate(new Date(ingestionFlowFileDTO.getCreationDate().getTime()).toLocalDate())
+			.organizationId(ingestionFlowFileDTO.getOrg().getOrgId())
+			.ingestionFlowFileId(ingestionFlowFileDTO.getIngestionFlowFileId())
+			.pspIdentifier(ctFlussoRiversamento.getIstitutoMittente().getDenominazioneMittente())
+			.flowIdentifierCode(ctFlussoRiversamento.getIdentificativoFlusso())
+			.flowDateTime(ctFlussoRiversamento.getDataOraFlusso().toGregorianCalendar().toZonedDateTime().toLocalDate())
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:40]: check if the field contains just date or date/time

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+			.regulationUniqueIdentifier(ctFlussoRiversamento.getIdentificativoUnivocoRegolamento())
+			.regulationDate(ctFlussoRiversamento.getDataRegolamento().toGregorianCalendar().toZonedDateTime().toLocalDate())
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:42]: check if the field contains just date or date/time

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+			.senderPspName(ctFlussoRiversamento.getIstitutoMittente().getDenominazioneMittente())
+			.senderPspCode(ctFlussoRiversamento.getIstitutoMittente().getIdentificativoUnivocoMittente().getCodiceIdentificativoUnivoco())
+			.senderPspType(ctFlussoRiversamento.getIstitutoMittente().getIdentificativoUnivocoMittente().getTipoIdentificativoUnivoco().value())
+			.receiverOrganizationName(ctFlussoRiversamento.getIstitutoRicevente().getDenominazioneRicevente())
+			.receiverOrganizationCode(ctFlussoRiversamento.getIstitutoRicevente().getIdentificativoUnivocoRicevente().getCodiceIdentificativoUnivoco())
+			.receiverOrganizationType(ctFlussoRiversamento.getIstitutoRicevente().getIdentificativoUnivocoRicevente().getTipoIdentificativoUnivoco().value())
+			.totalPayments(ctFlussoRiversamento.getDatiSingoliPagamenti().size())
+			.totalAmountCents(ctFlussoRiversamento.getImportoTotalePagamenti().movePointRight(2).longValueExact())
+			.totalPayments(ctFlussoRiversamento.getNumeroTotalePagamenti().longValueExact())
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:51]: you are setting this twice

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:1]: I see that you are reading nested fields without checking for null values, are these fields required? (we have a xsd that will guarantee it for us, have you checked it?)

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+			.bicCodePouringBank(ctFlussoRiversamento.getCodiceBicBancaDiRiversamento());
+
+		return ctFlussoRiversamento.getDatiSingoliPagamenti().stream()
+			.map(item -> builder
+				.creditorReferenceId(item.getIdentificativoUnivocoVersamento())
+				.regulationId(item.getIdentificativoUnivocoRiscossione())
+				.transferIndex(item.getIndiceDatiSingoloPagamento())
+				.amountPaidCents(item.getSingoloImportoPagato().movePointRight(2).longValueExact())
+				.paymentOutcomeCode(item.getCodiceEsitoSingoloPagamento())
+				.payDate(item.getDataEsitoSingoloPagamento().toGregorianCalendar().toZonedDateTime().toLocalDate())
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:61]: check if the field contains just date or date/time

File: src/test/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImplTest.java [hunk 26]
```diff
@@ -33,10 +41,15 @@ class PaymentsReportingIngestionFlowFileActivityImplTest {
...
 	private FlussoRiversamentoUnmarshallerService flussoRiversamentoUnmarshallerServiceMock;
+	@Mock
+	private PaymentsReportingIngestionFlowFileValidatorService paymentsReportingIngestionFlowFileValidatorServiceMock;
+	@Mock
+	private PaymentsReportingMapperService paymentsReportingMapperServiceMock;
 	private PaymentsReportingIngestionFlowFileActivityImpl ingestionActivity;
 
 	@TempDir
 	private File tempDir;
+	private CtFlussoRiversamento ctFlussoRiversamento;
```
Comment (antonioT90) [inline src/test/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImplTest.java:52]: why you have an instance variable, but you are calling the new on each test?

File: src/test/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperServiceTest.java [hunk 27]
```diff
@@ -0,0 +1,96 @@
...
+		assertEquals(1L, firstDTO.getTotalPayments());
+		assertEquals(100_050L, firstDTO.getTotalAmountCents());
+		assertEquals("vers123", firstDTO.getCreditorReferenceId());
+		assertEquals("ris123", firstDTO.getRegulationId());
+		assertEquals(1, firstDTO.getTransferIndex());
+		assertEquals(20_000L, firstDTO.getAmountPaidCents());
+		assertEquals("OK", firstDTO.getPaymentOutcomeCode());
+		assertNotNull(firstDTO.getPayDate());
+
+		TestUtils.checkNotNullFields(firstDTO, "creationDate", "lastChangeDate", "paymentsReportingId");
```
Comment (antonioT90) [inline src/test/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperServiceTest.java:90]: ok for the id null, but creationDate and lastChangeDate should be null?

File: src/main/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImpl.java
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImpl.java:1]: The main method is started to grow...
could we move:
* row from 50 to 54 into a new private method `findIngestionFlowFileRecord`
* row from 56 to 58 into a new private method `retrieveFile`
* row from 60 to 65 into a new private method `parseData`

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java (hunk 20 above)
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:35]: it just a LocalDate

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java (hunk 22 above)
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:40]: changed, using java.time.ZonedDateTime#toLocalDateTime

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java (hunk 23 above)
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:42]: it is just date

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java (hunk 24 above)
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:51]: i've fixed it in the next commit

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:1]: All fields are required except `it.gov.digitpa.schemas._2011.pagamenti.CtFlussoRiversamento#codiceBicBancaDiRiversamento` and `it.gov.digitpa.schemas._2011.pagamenti.CtDatiSingoliPagamenti#indiceDatiSingoloPagamento`

File: src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java (hunk 25 above)
Comment (macacia) [reply src/main/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperService.java:61]: it is just a date

File: src/test/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImplTest.java (hunk 26 above)
Comment (macacia) [reply src/test/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImplTest.java:52]: It just a typo, Ill' fix it

File: src/test/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperServiceTest.java (hunk 27 above)
Comment (macacia) [reply src/test/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperServiceTest.java:90]: removed, there's no default value

**Review by antonioT90 — CHANGES_REQUESTED** [review]

**Pull request merged**